aptos-move/framework/aptos-stdlib/sources/pool_u64.move

```diff
@@ -0,0 +1,535 @@
+/**
+ * Simple module for tracking and calculating shares of a pool of coins. The shares are worth more as the total coins in
+ * the pool increases. New shareholder can buy more shares or redeem their existing shares.
+ *
+ * Example flow:
+ * 1. Pool start outs empty.
+ * 2. Shareholder A buys in with 1000 coins. A will receive 1000 shares in the pool. Pool now has 1000 total coins and
+ * 1000 total shares.
+ * 3. Pool appreciates in value from rewards and now has 2000 coins. A's 1000 shares are now worth 2000 coins.
+ * 4. Shareholder B now buys in with 1000 coins. Since before the buy in, each existing share is worth 2 coins, B will
+ * receive 500 shares in exchange for 1000 coins. Pool now has 1500 shares and 3000 coins.
+ * 5. Pool appreciates in value from rewards and now has 6000 coins.
+ * 6. A redeems 500 shares. Each share is worth 6000 / 1500 = 4. A receives 2000 coins. Pool has 4000 coins and 1000
+ * shares left.
+ */
+module aptos_std::pool_u64 {
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use std::error;
+    use std::vector;
+
+    /// Shareholder not present in pool.
+    const ESHAREHOLDER_NOT_FOUND: u64 = 1;
+    /// There are too many shareholders in the pool.
+    const ETOO_MANY_SHAREHOLDERS: u64 = 2;
+    /// Cannot destroy non-empty pool.
+    const EPOOL_IS_NOT_EMPTY: u64 = 3;
+    /// Cannot redeem more shares than the shareholder has in the pool.
+    const EINSUFFICIENT_SHARES: u64 = 4;
+    /// Shareholder cannot have more than u64.max shares.
+    const ESHAREHOLDER_SHARES_OVERFLOW: u64 = 5;
+    /// Pool's total coins cannot exceed u64.max.
+    const EPOOL_TOTAL_COINS_OVERFLOW: u64 = 6;
+    /// Pool's total shares cannot exceed u64.max.
+    const EPOOL_TOTAL_SHARES_OVERFLOW: u64 = 7;
+
+    const MAX_U64: u64 = 18446744073709551615;
+
+    struct Pool has store {
+        shareholders_limit: u64,
+        total_coins: u64,
```

Comment:
should this be the actual coin? otherwise, how to prove this pool actually contains this amount of coin?